### PR TITLE
Add I2C General Call Commands

### LINKED ISSUE #9: Add I2C General Call Commands
we could add the following I2C General Call Commands, to enhance the features of this library
- General Call Reset
- General Call Wake-Up
- General Call Software Update
- General Call Read Address Bits

### PR DESCRIPTION
**Pull request by codenio**
This PR address #9 , 
- Adds _general_call static methods for executing different general calls to MCP4728
- Adds reset, wakeup and soft_update general call static methods

Thoughts and Comments..?

### CODE REVIEW

**Comment by jposada202020** [comment]
@codenio that was quick, questions:
Sorry not to mention that the CI will probably fail as we use pylint and Black to verify our code and sphinx for the documentation
please see. https://learn.adafruit.com/improve-your-code-with-pylint
Also question why you are putting the functions in the CV class, we only use this one as a constructor to implement options for the registers and others.
I will let run the CI and see what happens : Thanks

**Comment by jposada202020** [comment]
@codenio I will take a look at the datasheet tonight,  and try to test moving your functions to the main class, also a good practice is to have a test code, that way we could test the same things and compare results. Thanks...

**Comment by codenio** [comment]
> Sorry not to mention that the CI will probably fail as we use pylint and Black to verify our code and sphinx for the documentation
> please see. https://learn.adafruit.com/improve-your-code-with-pylint

Thanks for prompt response. adding the same in contribution guidelines would help new contributors.

Will make those suggested changes and update this PR

**Comment by jposada202020** [comment]
You are right, actually it is the guide, https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/overview. We are working on putting some automatic "guidelines" in the CI built. But, thank you, your suggestion is noted.

**Comment by codenio** [comment]
> try to test moving your functions to the main class, 

I guess it is in main class "MCP4728", could you check the same..?

> also a good practice is to have a test code, that way we could test the same things and compare results. Thanks...

test code you mean, example code to test it on actual hardware..?
I can add the example code but I do not have an MCP4728 BoB with me, any help in testing this code is appriciated

Thanks
codenio

**Comment by jposada202020** [comment]
You are right, just to verify that the new features introduced with your Pr are working, I Do have the hardware so I could test :)

**Comment by codenio** [comment]
The following changes were made
- changed newly added functions as class methods (instead of static methods)
- added example file to test the reset function
- used Pre-commit to lint the code

will modify the implementation to improve readability and get back

**Comment by codenio** [comment]
All changes done. hope this could be reviewed and merged. Thanks for the prompt responses and suggestions.

**Comment by jposada202020** [comment]
@codenio just to let you know, that I took a briefly look at the changes and test some of the new features, it works. will take some time later to give my official review. Thanks for your contribution

File: examples/mcp4728_generalcalltest.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# SPDX-FileCopyrightText: 2021 ladyada for Adafruit Industries
```
Comment (jposada202020) [inline examples/mcp4728_generalcalltest.py]: You could change this with you name or your github user if you like

**Review by jposada202020 — CHANGES_REQUESTED** [review]
Thanks just that minor comment regarding the coyright mention, also we need to include this new example in the examples.rst file, following the standard provides, this allows the autodocs to build the readthedocs documentation

I have testes this with a modified script provided, using the internal function `_read_registers()` to see changes.

## Modified Test Code
```python
import board
import adafruit_mcp4728

i2c = board.I2C()  # uses board.SCL and board.SDA
mcp4728 = adafruit_mcp4728.MCP4728(i2c)
print(mcp4728._read_registers())
mcp4728.channel_a.value = 65535  # Voltage = VDD
mcp4728.channel_b.value = int(65535 / 2)  # VDD/2
mcp4728.channel_c.value = int(65535 / 4)  # VDD/4
mcp4728.channel_d.value = 0  # 0V

mcp4728.save_settings()  # save current voltages into EEPROM
print(mcp4728._read_registers())
mcp4728.channel_a.value = 0  # Modify output
mcp4728.channel_b.value = 0  # Modify output
mcp4728.channel_c.value = 0  # Modify output
mcp4728.channel_d.value = 65535  # Modify output

print(mcp4728._read_registers())
mcp4728.reset()  # reset MCP4728
print(mcp4728._read_registers())
```

## Results
```
Adafruit CircuitPython 6.2.0 on 2021-04-05; Adafruit Feather RP2040 with rp2040
>>> import mcp4728_generaltest
[(6, False, False, 0), (2047, False, False, 0), (1023, False, False, 0), (0, False, False, 0)]
[(4095, False, False, 0), (2047, False, False, 0), (1023, False, False, 0), (0, False, False, 0)]
[(4095, False, False, 0), (2047, False, False, 0), (1023, False, False, 0), (0, False, False, 0)]
[(6, False, False, 0), (2047, False, False, 0), (1023, False, False, 0), (0, False, False, 0)]
>>> 
```

Changes seems good to me

File: examples/mcp4728_generalcalltest.py (hunk 1 above)
Comment (codenio) [reply examples/mcp4728_generalcalltest.py]: made those suggested changes

**Comment by jposada202020** [comment]
@codenio sorry it was not my intention to try to imply that we needed to change the example as you made it. Just to clarify to folks how I did my tests and verify the changes. Your code was perfectly fine to begin with

**Comment by codenio** [comment]
> not my intention to try to imply that we needed to change the example as you made it.

realised the same just now, will revert to previous version of the example.
BTW, do we need to expose _read_registers() ..? is it a valid feature..?

**Comment by jposada202020** [comment]
No we are good, user will not need that feature, and if they needed they will use as we did :)

**Comment by jposada202020** [comment]
you could correct the last one with this

```python
General Call Example
------------------------------

````

**Comment by codenio** [comment]
> you could correct the last one with this
> 
> ```python
> General Call Example
> ------------------------------
> ```

is there way to check this before pushing..?

**Comment by jposada202020** [comment]
Yes, there is, you could run the following command 
`sphinx-build -E -W -b html . _build/html`
This will build the docs as the CI does

**Comment by codenio** [comment]
@jposada202020 , all these short debugging commands can be add into contribution guidelines =)

**Comment by jposada202020** [comment]
@codenio :) I know, I am on it :), I open this discussion in the CircuitPython dev channel that by the way you are invited to participate, is open, for everyone is in the Adafruit discord channel under circuitpython-dev.

**Review by jposada202020 — APPROVED** [review]

**Comment by jposada202020** [comment]
Thanks to stick to the whole process, know we give folks a couple of days and then we could merge. Thanks :)

**Comment by codenio** [comment]
thanks for the prompt responses and guidance :+1:

**Pull request merged**